tests/src/tests/intents/swap_ft/mod.rs

```diff
@@ -0,0 +1,898 @@
+    assert_eq!(
+        env.token_a
+            .as_account()
+            .ft_balance_of(env.user_id())
+            .await
+            .unwrap(),
+        1000,
+    );
+    assert_eq!(
+        env.token_b
+            .as_account()
+            .ft_balance_of(env.user_id())
+            .await
+            .unwrap(),
+        0
+    );
+
+    assert_eq!(
+        env.token_a
+            .as_account()
+            .ft_balance_of(env.solver_id())
+            .await
+            .unwrap(),
+        0
+    );
+    assert_eq!(
+        env.token_b
+            .as_account()
+            .ft_balance_of(env.solver_id())
+            .await
+            .unwrap(),
+        2000
+    );
+
+    // User creates intent for swapping 1000 TokenA to 2000 TokenB.
+    env.user
+        .create_intent(
+            env.token_a.id(),
+            env.ft_intent.id(),
+            "1",
+            Intent {
+                initiator: env.user_id().clone(),
+                send: TokenAmount {
+                    token_id: env.token_a.id().clone(),
+                    amount: 1000.into(),
+                },
+                receive: TokenAmount {
+                    token_id: env.token_b.id().clone(),
+                    amount: 2000.into(),
+                },
+                expiration: Expiration::default(),
+                referral: None,
+            },
+        )
+        .await;
+
+    // Check that intent contract owns user's TokenA.
+    assert_eq!(
+        env.token_a
+            .as_account()
+            .ft_balance_of(env.user_id())
+            .await
+            .unwrap(),
+        0
+    );
+    assert_eq!(
+        env.token_a
+            .as_account()
+            .ft_balance_of(env.ft_intent.id())
+            .await
+            .unwrap(),
+        1000
+    );
+
+    // The solver is happy with such intent and executes it.
+    env.solver
+        .execute_intent(env.token_b.id(), env.ft_intent.id(), "1", 2000.into())
+        .await;
+
+    // Check balances after intent execution.
+    assert_eq!(
+        env.token_a
+            .as_account()
+            .ft_balance_of(env.user_id())
+            .await
+            .unwrap(),
+        0
+    );
+    assert_eq!(
+        env.token_b
+            .as_account()
+            .ft_balance_of(env.user_id())
+            .await
+            .unwrap(),
+        2000
+    );
+
+    assert_eq!(
+        env.token_a
+            .as_account()
+            .ft_balance_of(env.solver_id())
+            .await
+            .unwrap(),
+        1000
+    );
+    assert_eq!(
+        env.token_b
+            .as_account()
+            .ft_balance_of(env.solver_id())
+            .await
+            .unwrap(),
+        0
+    );
+
+    // Check that intent has been removed form the state.
+    let intent = env.user.get_intent(env.ft_intent.id(), "1").await.unwrap();
+    assert!(matches!(intent.status(), Status::Completed));
+}
```

Comment:
Could you please make the tests compact as they were? It's too hard to read them and understand what it about now.